iina/OpenSubSubtitle.swift

```diff
@@ -212,28 +220,29 @@ class OpenSubSupport {
 
       hash += fileSize
 
-      file.closeFile()
-
       resolver.fulfill(FileInfo(hashValue: String(format: "%016qx", hash), fileSize: fileSize))
     }
   }
 
-  func requestByName(_ fileURL: URL) -> Promise<[OpenSubSubtitle]> {
-    return requestIMDB(fileURL).then { imdb -> Promise<[OpenSubSubtitle]> in
+  func requestByName(_ fileURL: URL, _ playerCore: PlayerCore) -> Promise<[OpenSubSubtitle]> {
+    return requestIMDB(fileURL, playerCore).then { imdb -> Promise<[OpenSubSubtitle]> in
       let info = ["imdbid": imdb]
       return self.request(info)
     }
   }
 
-  func requestIMDB(_ fileURL: URL) -> Promise<String> {
+  func requestIMDB(_ fileURL: URL, _ playerCore: PlayerCore) -> Promise<String> {
```

Comment:
I think a better choice would be to ask the caller to provide a fallback title for the URL.

Reply:
Fixed.